### PR TITLE
(3.0) txn: Check lock's TTL when doing cleanup (#5471)

### PR DESCRIPTION
**Pull request by MyonKeminta**
###  What have you changed?

This PR cherry-picks #5471 , which is part of refinement of pessimistic transactions, to release-3.0 branch. For more details please view #5471 .

###  What is the type of the changes?

- New feature (a change which adds functionality)

###  How is the PR tested?

- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

- No.

###  Does this PR affect `tidb-ansible`?

No.

###  Refer to a related PR or issue link (optional)

* Cherry-picks: #5471 
* Corresponding PR in TiDB: pingcap/tidb#12417

### CODE REVIEW

**Comment by MyonKeminta** [comment]
Also PTAL @coocood @tiancaiamao

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -486,8 +486,34 @@ impl<S: Snapshot> MvccTxn<S> {
...
+            Some(ref mut lock) if lock.ts == self.start_ts => {
+                // If current_ts is not 0, check the Lock's TTL.
+                // If the lock is not expired, do not rollback it but report key is locked.
+                if current_ts > 0
+                    && extract_physical(lock.ts) + lock.ttl >= extract_physical(current_ts)
+                {
+                    // The `lock.primary` field will not be accessed again. Use mem::replace to
+                    // avoid cloning.
+                    let primary = ::std::mem::replace(&mut lock.primary, Default::default());
+                    return Err(Error::KeyIsLocked {
```
Comment (coocood) [inline src/storage/mvcc/txn.rs]: Why these lines are different? https://github.com/tikv/tikv/pull/5471/files#diff-10ac7ebd80ca4da7cccc8e3539f12eb4R507

Comment (MyonKeminta) [reply src/storage/mvcc/txn.rs]: Because the definition of `Error::KeyIsLocked` was changed in #5152 , which is not included in release-3.x branches.

**Comment by coocood** [comment]
LGTM

**Pull request merged**